lighthouse-core/scripts/dogfood-lhci.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -euox pipefail
+
+# This script requires LHCI_CANARY_SERVER_URL and LHCI_CANARY_SERVER_TOKEN variables to be set.
+
+if [[ "$TRAVIS_NODE_VERSION" != "10" ]]; then
+  echo "Not running dogfood script on node versions other than 10";
+  exit 0;
+fi
+
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+LH_ROOT_DIR="$SCRIPT_DIR/../.."
+
+# Testing lhci takes time and the server ain't massive, we'll only run the tests if we touched files that affect the report.
+CHANGED_FILES=""
+if [[ "$CI" ]]; then
+  CHANGED_FILES=$(git --no-pager diff --name-only "$TRAVIS_COMMIT_RANGE")
+else
+  CHANGED_FILES=$(git --no-pager diff --name-only master)
+fi
+
+printf "Determined the following files have been touched:\n\n$CHANGED_FILES\n\n"
+
+if ! echo "$CHANGED_FILES" | grep -E 'report|lhci' > /dev/null; then
+  echo "No report files affected, skipping lhci checks."
+  exit 0
+fi
+
+# Generate an HTML report
+node "$LH_ROOT_DIR/lighthouse-cli" \
+  -A="$LH_ROOT_DIR/lighthouse-core/test/results/artifacts" \
+  --config-path="$LH_ROOT_DIR/lighthouse-core/test/results/sample-config.js" \
+  --output-path="lhci-report.html" \
+  --quiet --output=html
+
+# Start up a test server.
+npx http-server -p 9090 &
```

Comment:
is using static-server too much of a pain?

or could we use python simplehttpserver?

Reply:
`python3 -m http.server 9000` works too

Reply:
did not expect such controversy over it haha, this is how I would've expected documenting this to a potential lhci user who might only have node in their environment so wanted to use it like they would.

I can switch to static-server.js if we'd prefer

Reply:
Does that mean we'd be adding static-server and its features to our stable API?

Reply:
> Does that mean we'd be adding static-server and its features to our stable API?

I think the intention is for this to just test out lhci on the report, not be a general solution for folks using lhci (whether or not it should be written like the general solution is another question, but I guess this has the benefit of not needing another dep or download?)

Reply:
> I think the intention is for this to just test out lhci on the report, not be a general solution for folks using lhci (whether or not it should be written like the general solution is another question, but I guess this has the benefit of not needing another dep or download?)

Yeah exactly.

> Does that mean we'd be adding static-server and its features to our stable API?

Nope, this script has absolutely no bearing on the API for lighthouse or lhci. It's just about how we use lhci for our internal testing.